Data race in modules integration

In current `master`:

```
$ go test -race -run Concurrent
==================
WARNING: DATA RACE
Read at 0x000000e78b87 by goroutine 10:
  github.com/getsentry/sentry-go.extractModules()                                                
      /home/rodolfo/git/sentry-go/integrations.go:41 +0x3e 
[...]
Previous write at 0x000000e78b87 by goroutine 12:                                                
  github.com/getsentry/sentry-go.extractModules()                                                
      /home/rodolfo/git/sentry-go/integrations.go:45 +0x5b   
[...]
==================
--- FAIL: TestConcurrentHubClone (0.00s)
    testing.go:853: race detected during execution of test
FAIL
exit status 1
FAIL	github.com/getsentry/sentry-go	0.022s
```

Note that the data race is not detected when all tests are run with `go test -race`.


```
$ git log --oneline -1                                                                                          
54d5222 (HEAD -> master, upstream/master) feat: Add support for path prefixes in the DSN (#102)
$ git log --oneline -S TestConcurrentHubClone                                                                   
58c0cb1 test: Add test for concurrent Hub cloning
$ git bisect start 54d5222 58c0cb1 --                                
$ git bisect run go test -race -run '^TestConcurrentHubClone$'
...
17042b22bdd50932c75dc5216f7d422d6bcb8382 is the first bad commit
commit 17042b22bdd50932c75dc5216f7d422d6bcb8382 
Author: Chris Faulkner <thefaulkner@gmail.com>
Date:   Tue Oct 8 01:23:05 2019 -0700

    fix: Handle new lines in module integration (#65)

    NewEvent() doesn't return Modules as nil, which was preventing this integration from working.
    The extra flag was added to prevent repeatedly trying to cache the modules if getModules() returned nil.
    
 integrations.go | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)
bisect run success 
```

Regression bisected to 17042b22bdd50932c75dc5216f7d422d6bcb8382.

Affected lines [integrations.go:41](https://github.com/getsentry/sentry-go/blob/54d52224adc5e7cba27907d9193de18724f4ee59/integrations.go#L41) and [integrations.go:45](https://github.com/getsentry/sentry-go/blob/54d52224adc5e7cba27907d9193de18724f4ee59/integrations.go#L45):
https://github.com/getsentry/sentry-go/blob/54d52224adc5e7cba27907d9193de18724f4ee59/integrations.go#L40-L55

--- a/integrations.go
+++ b/integrations.go
@@ -16,10 +16,10 @@ import (
 // Modules Integration
 // ================================
 
-type modulesIntegration struct{}
-
-var _modulesCache map[string]string // nolint: gochecknoglobals
-var _modulesCached bool             // nolint: gochecknoglobals
+type modulesIntegration struct {
+	once    sync.Once
+	modules map[string]string
+}
 
 func (mi *modulesIntegration) Name() string {
 	return "Modules"
@@ -31,26 +31,20 @@ func (mi *modulesIntegration) SetupOnce(client *Client) {
 
 func (mi *modulesIntegration) processor(event *Event, hint *EventHint) *Event {
 	if len(event.Modules) == 0 {
-		event.Modules = extractModules()
+		mi.once.Do(func() {
+			mi.modules = extractModules()
+		})
 	}
-
+	event.Modules = mi.modules
 	return event
 }
 
 func extractModules() map[string]string {
-	if _modulesCached {
-		return _modulesCache
-	}
-
-	_modulesCached = true
 	extractedModules, err := getModules()
 	if err != nil {
 		Logger.Printf("ModuleIntegration wasn't able to extract modules: %v\n", err)
 		return nil
 	}
-
-	_modulesCache = extractedModules
-
 	return extractedModules
 }
 

